core/src/main/java/org/infinispan/scattered/impl/ScatteredStateConsumerImpl.java

```diff
@@ -239,11 +239,11 @@ protected void onTaskCompletion(InboundTransferTask inboundTransfer) {
                   svm.notifyKeyTransferFinished(segment, inboundTransfer.isCompletedSuccessfully(), inboundTransfer.isCancelled());
                   switch (completedSegments.size()) {
                      case 0:
-                        completedSegments = Collections.singleton(segment);
+                        completedSegments = IntSets.immutableSet(segment);
                         break;
                      case 1:
-                        completedSegments = new HashSet<>(completedSegments);
-                        // intentional no break
+                        completedSegments = IntSets.mutableCopyFrom(completedSegments);
+                        // Intentional no break
```

Comment:
Since you're changing this, I suggest a "fallthrough" comment to placate IntelliJ's inspection. I'm not going to go into how could the set of completed segments ever differ from the transfer set of segments, if there's always a 1-to-1 mapping between segments and transfers :)

Reply:
Ah I don't have that inspection enabled, as it is disabled by default. Fixed the comment though

https://stackoverflow.com/questions/43395722/intellij-inspection-for-switch-case-fall-through